frontend-ssr/components/tableOfContents/TableOfContents.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import styles from './TableOfContents.module.scss';
+
+interface State {
+    activeChapterId: string,
+}
+
+interface Props {
+    chapters: Chapters[],
+}
+
+interface Chapters {
+    title: string,
+    id: string,
+}
+
+class TableOfContents extends React.Component<Props, State> {
+  private observer: any;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeChapterId: props.chapters[0].id,
+    };
+  }
+
+  componentDidMount = () => {
+    const changeNav = (entries) => {
+      entries.forEach((entry) => {
+        let entryId = document.getElementById(entry.target.id);
+        if (entryId !== null && entry.intersectionRatio > 0.5) {
+          // remove old active class
+          document.querySelector(`.${styles.active}`).classList.remove(`${styles.active}`);
+          // find matching link & add active class
+          document.querySelector(`[href="#${entry.target.id}"]`).classList.add(`${styles.active}`);
+        }
+      });
+    };
+
+    const options = {
+      threshold: 0.5,
+    };
+
+    this.observer = new IntersectionObserver(changeNav, options);
+
+    // target the elements to be observed
+    const { chapters } = this.props;
+    chapters.forEach((item) => {
+      const chapterId = document.getElementById(item.id);
+      this.observer.observe(chapterId);
+    });
+  }
+
+  componentWillUnmount = () => {
+    this.observer.disconnect();
+  }
+
+  handleScrollItem = (id) => {
+    let itemToScroll = document.getElementById(id);
+    if (itemToScroll === null) {
+      console.warn('TableOfContents: expected to find element with id="id" to scroll to');
+    }
+    this.observer.unobserve(itemToScroll);
```

Comment:
Here we need to unobserve all elements. It's not enough if we do just for this one, unfortunately.

Also, please add an explainer regarding what's the purpose of this, what problem we're trying to solve., for future reference.